src/kademlia/rpc_manager.cpp

```diff
@@ -444,6 +417,12 @@ bool rpc_manager::invoke(entry& e, udp::endpoint target_addr
 	// places a 'ro' key in the top-level message dictionary and sets its value to 1.
 	if (m_settings.read_only) e["ro"] = 1;
 
+	node& n = o->algorithm()->get_node();
+	if (!n.native_address(o->target_addr()))
```

Comment:
ah, native_address() means is_native_address().. it took me a minute to make sense of this code